Interfaces/services/twitter_service.py

```diff
@@ -0,0 +1,176 @@
+            self.ACCESS_TOKEN: "",
+            self.ACCESS_TOKEN_SECRET: ""
+        }
+
+    def get_required_config(self):
+        return [self.API_KEY, self.API_SECRET, self.ACCESS_TOKEN, self.ACCESS_TOKEN_SECRET]
+
+    @classmethod
+    def get_help_page(cls) -> str:
+        return "https://github.com/Drakkar-Software/OctoBot/wiki/Twitter-interface#twitter-interface"
+
+    @staticmethod
+    def is_setup_correctly(config):
+        return CONFIG_TWITTER in config[CONFIG_CATEGORY_SERVICES] \
+               and CONFIG_SERVICE_INSTANCE in config[CONFIG_CATEGORY_SERVICES][CONFIG_TWITTER]
+
+    def get_user_id(self, user_account):
+        user = self.twitter_api.GetUser(screen_name=user_account)
+        return user.id
+
+    def get_history(self, user_id):
+        return self.twitter_api.GetUserTimeline(user_id=user_id)
+
+    async def prepare(self):
+        if not self.twitter_api:
+            self.twitter_api = twitter.Api(
+                self.config[CONFIG_CATEGORY_SERVICES][CONFIG_TWITTER][self.API_KEY],
+                self.config[CONFIG_CATEGORY_SERVICES][CONFIG_TWITTER][self.API_SECRET],
+                self.config[CONFIG_CATEGORY_SERVICES][CONFIG_TWITTER][self.ACCESS_TOKEN],
+                self.config[CONFIG_CATEGORY_SERVICES][CONFIG_TWITTER][self.ACCESS_TOKEN_SECRET],
+                sleep_on_rate_limit=True
+            )
+
+    def get_type(self):
+        return CONFIG_TWITTER
+
+    def get_endpoint(self):
+        return self.twitter_api
+
+    def has_required_configuration(self):
+        return CONFIG_CATEGORY_SERVICES in self.config \
+               and CONFIG_TWITTER in self.config[CONFIG_CATEGORY_SERVICES] \
+               and self.check_required_config(self.config[CONFIG_CATEGORY_SERVICES][CONFIG_TWITTER])
+
+    @staticmethod
+    def decode_tweet(tweet):
+        if "extended_tweet" in tweet and "full_text" in tweet:
+            return tweet["extended_tweet"]["full_text"]
+        elif "text" in tweet:
+            return tweet["text"]
+        return ""
+
+    async def post(self, content, error_on_failure=True):
+        try:
+            return await self.split_tweet_content(content=content, tweet_id=None)
+        except Exception as e:
+            error = f"Failed to send tweet : {e} tweet:{content}"
+            if error_on_failure:
+                self.logger.error(error)
+            else:
+                self.logger.info(error)
+        return None
+
+    async def respond(self, tweet_id, content, error_on_failure=True):
+        try:
+            return await self.split_tweet_content(content=content, tweet_id=tweet_id)
+        except Exception as e:
+            error = f"Failed to send response tweet : {e} tweet:{content}"
+            if error_on_failure:
+                self.logger.error(error)
+            else:
+                self.logger.info(error)
+        return None
+
+    async def split_tweet_content(self, content, counter=None, counter_max=None, tweet_id=None):
```

Comment:
I don't understand why this method has to be async ?

Reply:
legacy issue, I'll fix it and rename the method at the same time, it splits and sends, not just split

Reply:
Nice :+1: